cinema/views.py

```diff
@@ -56,3 +90,48 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie_id:
+            queryset = queryset.filter(movie_id=int(movie_id))
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=(
```

Comment:
If you start with "tickets_available=(", u must finish the same way with ")". Not "- Count("tickets")))"

I hope u understand)))

Reply:
Also I prefer it to be calculated field, to provide access on all layers. But it's not mandatory.